client/layout/guided-tours/steps.js

```diff
@@ -11,22 +11,40 @@ import Card from 'components/card';
 import Button from 'components/button';
 import ExternalLink from 'components/external-link';
 import Gridicon from 'components/gridicon';
-import { posToCss, getStepPosition, getBullseyePosition, targetForSlug } from './positioning';
+import { posToCss, getStepPosition, getBullseyePosition, getOverlayStyle, getScrollDiff, targetForSlug, getScrolledRect, query } from './positioning';
 
 class BasicStep extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { stepPosition: null, targetRect: null };
+	}
+
+	componentWillMount() {
+		const container = query( '#secondary .sidebar' )[ 0 ];
+		const scrollY = getScrollDiff( this.props.targetSlug, container );
+		const targetRect = getScrolledRect( { targetSlug: this.props.targetSlug, scrollY: scrollY } );
+
+		this.setState( { stepPosition: posToCss( getStepPosition( this.props ) ) } );
+		this.setState( { targetRect: targetRect } );
```

Comment:
> Neat--but isn't it more easily readable when keeping it explicit?

That's likely subjective. For myself, it's very natural to use shorthand for these situations, and I use/see it often throughout Calypso.